py-polars/polars/expr/binary.py

```diff
@@ -30,30 +30,128 @@ def contains(self, literal: bytes) -> Expr:
         -------
         Boolean mask
 
+        See Also
+        --------
+        starts_with : Check if the binary substring exists at the start
+        ends_with : Check if the binary substring exists at the end
+
+        Examples
+        --------
+        >>> colors = pl.DataFrame(
+        ...     {
+        ...         'name': ['black', 'yellow', 'blue'],
+        ...         'code': [b'\x00\x00\x00', b'\xff\xff\x00', b'\x00\x00\xff']
+        ...     }
+        ... )
+        >>> colors.select(
+        ...     'name',
+        ...     pl.col('code').bin.encode('hex').alias('code_encoded_hex'),
+        ...     pl.col('code').bin.contains(b'\xff').alias('contains_ff'),
+        ...     pl.col('code').bin.starts_with(b'\xff').alias('starts_with_ff'),
+        ...     pl.col('code').bin.ends_with(b'\xff').alias('ends_with_ff'),
+        ... )
+        shape: (3, 5)
+        ┌────────┬──────────────────┬─────────────┬────────────────┬──────────────┐
+        │ name   ┆ code_encoded_hex ┆ contains_ff ┆ starts_with_ff ┆ ends_with_ff │
+        │ ---    ┆ ---              ┆ ---         ┆ ---            ┆ ---          │
+        │ str    ┆ str              ┆ bool        ┆ bool           ┆ bool         │
+        ╞════════╪══════════════════╪═════════════╪════════════════╪══════════════╡
+        │ black  ┆ 000000           ┆ false       ┆ false          ┆ false        │
+        │ yellow ┆ ffff00           ┆ true        ┆ true           ┆ false        │
+        │ blue   ┆ 0000ff           ┆ true        ┆ false          ┆ true         │
+        └────────┴──────────────────┴─────────────┴────────────────┴──────────────┘
         """
         return wrap_expr(self._pyexpr.bin_contains(literal))
 
     def ends_with(self, suffix: bytes) -> Expr:
-        """
+        r"""
         Check if string values end with a binary substring.
 
         Parameters
         ----------
         suffix
             Suffix substring.
 
+        Returns
+        -------
+        Boolean mask
+
+        See Also
+        --------
+        starts_with : Check if the binary substring exists at the start
+        contains : Check if the binary substring exists anywhere
+
+        Examples
+        --------
+        >>> colors = pl.DataFrame(
+        ...     {
+        ...         'name': ['black', 'yellow', 'blue'],
+        ...         'code': [b'\x00\x00\x00', b'\xff\xff\x00', b'\x00\x00\xff']
```

Comment:
Looks like this is missing a trailing comma. Try running `blackdoc`, that should fix it.